Create table LIKE another table not supported

create table temp.abc2 (like temp.abc1) ;
This is a valid Postgresql query but JsqlParser throws below exception :

Exception in thread "main" net.sf.jsqlparser.JSQLParserException
	at net.sf.jsqlparser.parser.CCJSqlParserUtil.parse(CCJSqlParserUtil.java:54)
Caused by: net.sf.jsqlparser.parser.ParseException: Encountered " "LIKE" "like "" at line 1, column 25.

--- a/src/main/java/net/sf/jsqlparser/statement/create/table/CreateTable.java
+++ b/src/main/java/net/sf/jsqlparser/statement/create/table/CreateTable.java
@@ -29,6 +29,7 @@ public class CreateTable implements Statement {
     private List<ColumnDefinition> columnDefinitions;
     private List<Index> indexes;
     private Select select;
+    private Table likeTable;
     private boolean selectParenthesis;
     private boolean ifNotExists = false;
     private RowMovement rowMovement;
@@ -106,6 +107,15 @@ public class CreateTable implements Statement {
         this.selectParenthesis = parenthesis;
     }
 
+    public Table getLikeTable() {
+        return likeTable;
+    }
+
+    public void setLikeTable(Table likeTable, boolean parenthesis) {
+        this.likeTable = likeTable;
+        this.selectParenthesis = parenthesis;
+    }
+
     public boolean isIfNotExists() {
         return ifNotExists;
     }
@@ -159,6 +169,9 @@ public class CreateTable implements Statement {
         }
         if (select != null) {
             sql += " AS " + (selectParenthesis ? "(" : "") + select.toString() + (selectParenthesis ? ")" : "");
+        }
+        if (likeTable != null) {
+            sql += " LIKE " + (selectParenthesis ? "(" : "") + likeTable.toString() + (selectParenthesis ? ")" : "");
         }
         return sql;
     }

--- a/src/main/java/net/sf/jsqlparser/util/deparser/CreateTableDeParser.java
+++ b/src/main/java/net/sf/jsqlparser/util/deparser/CreateTableDeParser.java
@@ -11,6 +11,7 @@ package net.sf.jsqlparser.util.deparser;
 
 import java.util.Iterator;
 
+import net.sf.jsqlparser.schema.Table;
 import net.sf.jsqlparser.statement.create.table.ColumnDefinition;
 import net.sf.jsqlparser.statement.create.table.CreateTable;
 import net.sf.jsqlparser.statement.create.table.Index;
@@ -91,6 +92,17 @@ public class CreateTableDeParser extends AbstractDeParser<CreateTable> {
             }
             Select sel = createTable.getSelect();
             sel.accept(this.statementDeParser);
+            if (createTable.isSelectParenthesis()) {
+                buffer.append(")");
+            }
+        }
+        if (createTable.getLikeTable() != null) {
+            buffer.append(" LIKE ");
+            if (createTable.isSelectParenthesis()) {
+                buffer.append("(");
+            }
+            Table table = createTable.getLikeTable();
+            buffer.append(table.getFullyQualifiedName());
             if (createTable.isSelectParenthesis()) {
                 buffer.append(")");
             }

--- a/src/main/jjtree/net/sf/jsqlparser/parser/JSqlParserCC.jjt
+++ b/src/main/jjtree/net/sf/jsqlparser/parser/JSqlParserCC.jjt
@@ -4048,6 +4048,7 @@ CreateTable CreateTable():
     List<String> idxSpec = new ArrayList<String>();
     Table fkTable = null;
     Select select = null;
+    Table likeTable = null;
     CheckConstraint checkCs = null;
     ExcludeConstraint excludeC = null;
     RowMovement rowMovement = null;
@@ -4200,6 +4201,10 @@ CreateTable CreateTable():
     [
     <K_AS> ( LOOKAHEAD("(" Select() ")") "(" select = Select() { createTable.setSelect(select, true); } ")"
              | select = Select() { createTable.setSelect(select, false); } )
+    ]
+    [
+    <K_LIKE> ( LOOKAHEAD("(" Table() ")") "(" likeTable=Table() { createTable.setLikeTable(likeTable, true); } ")"
+                 | likeTable=Table() { createTable.setLikeTable(likeTable, false); } )
     ]
     {
         createTable.setTable(table);

--- a/src/test/java/net/sf/jsqlparser/statement/create/CreateTableTest.java
+++ b/src/test/java/net/sf/jsqlparser/statement/create/CreateTableTest.java
@@ -670,4 +670,10 @@ public class CreateTableTest {
         String statement = "CREATE TABLE a LIKE b";
         assertSqlCanBeParsedAndDeparsed(statement);
     }
+
+    @Test
+    public void testCreateTableWithCommentIssue413_2() throws JSQLParserException {
+        String statement = "CREATE TABLE a LIKE (b)";
+        assertSqlCanBeParsedAndDeparsed(statement);
+    }
 }

--- a/src/test/java/net/sf/jsqlparser/statement/create/CreateTableTest.java
+++ b/src/test/java/net/sf/jsqlparser/statement/create/CreateTableTest.java
@@ -664,4 +664,10 @@ public class CreateTableTest {
         String sql = "CREATE TABLE test (startdate DATE) DISABLE ROW MOVEMENT AS SELECT 1 FROM dual";
         assertSqlCanBeParsedAndDeparsed(sql);
     }
+
+    @Test
+    public void testCreateTableWithCommentIssue413() throws JSQLParserException {
+        String statement = "CREATE TABLE a LIKE b";
+        assertSqlCanBeParsedAndDeparsed(statement);
+    }
 }

